internal/daemon/api_checks.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"net/http"
+
+	"github.com/canonical/pebble/internal/plan"
+	"github.com/canonical/pebble/internal/strutil"
+)
+
+type checkInfo struct {
+	Name     string `json:"name"`
+	Level    string `json:"level,omitempty"`
+	Healthy  bool   `json:"healthy"`
+	Failures int    `json:"failures,omitempty"`
+}
+
+func v1GetChecks(c *Command, r *http.Request, _ *userState) Response {
+	level := plan.CheckLevel(r.URL.Query().Get("level"))
+	switch level {
+	case plan.UnsetLevel, plan.AliveLevel, plan.ReadyLevel:
+	default:
+		return statusBadRequest(`level must be "alive" or "ready"`)
+	}
+
+	names := r.URL.Query()["names"]
```

Comment:
Hmmm, I see I've diverged here from `/v1/services`. We use both the comma-separate approach and specifying the query parameter multiple times. The current state of multi-valued query parameters is:

1. `GET /v1/files?action=read`: `path` can be specified multiple times (param name is singular)
2. `GET /v1/logs`: `services` can be specified multiple times (param name is plural)
3. `GET /v1/services`: `names` is a `CommaSeparatedList`

I believe quite strongly that specifying the query parameter multiple times is superior -- it's already an RFC-spec supported way to a enable multi-valued parameters, and there are no escaping issues with comma. Comma-separated lists, on the other hand, don't support having `,` in the value ... unless we were to define a way to escape, and then that would be a mini spec in itself, and one that only we use.

For example, in the `/v1/files` endpoint, a `path` might well have commas in it. Normal query string escaping applies, so these are automatically handled correctly.

So apart from the fact that even in existing code the multi-param approach is more common than the comma-separated-list approach, I believe it's saner and will avoid escaping issues down the track, so **I think we should continue with the multi-param approach.**

Minor detail: we should also probably settle on singular vs plural for the parameter name -- I lean towards plural as it gives you the hint that multiple of these can be specified (then again, this is quite debatable, you could say that you should use singular as each time you only specify one).

So I'll stick with the multi-param approach here. I'm happy to open an issue for `/v1/services` to allow `names` to be specified more than once too, to make that consistent moving forward (it would still split each param on comma for backwards-compatibility).

Reply:
Okay, per your comment below, I've added a `strutil.MultiCommaSeparatedList` function to support both methods for `/v1/checks?names=...`, `/v1/health?names=...`, `/v1/logs?services=...`, and `/v1/services?names=...`. That's all the existing cases except for the `/v1/files` one which needs to remain as is in case there is a `,` in the path.